toolchain/sem_ir/inst_profile.h

```diff
@@ -0,0 +1,21 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_SEM_IR_INST_PROFILE_H_
+#define CARBON_TOOLCHAIN_SEM_IR_INST_PROFILE_H_
+
+#include "llvm/ADT/FoldingSet.h"
+#include "toolchain/sem_ir/file.h"
+#include "toolchain/sem_ir/inst.h"
+
+namespace Carbon::SemIR {
+
+// Computes a profile of the given constant instruction, for canonicalization /
+// deduplication.
+auto ProfileConstant(llvm::FoldingSetNodeID& id, File* sem_ir, Inst inst)
```

Comment:
Sorry, too much thinking in Carbon :) (I was passing `FoldingSetNodeID` by reference because that's idiomatic and forced upon us in various places by the LLVM API, but otherwise defaulting to using pointers because that's how we'd do it in Carbon.)

Switched to references for parameters and pointers for class members.